### PR TITLE
track metadata changes

### PR DESCRIPTION
**Pull request by phoenixAja**
### Summary:
- **What:** add the ability to track when the user makes metadata changes
- **Ticket:** no ticket, mostly internal things to make the whole update flow work
- **Env:** no rdev since changes are non user facing

### Checklist:
- [x] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [x] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/frontend/src/components/WebformTable/index.tsx
Comment (mmmmmmaya) [inline src/frontend/src/components/WebformTable/index.tsx:26]: is the type really unknown? wouldn't it mainly be string or boolean?

File: src/frontend/src/views/Upload/components/Metadata/index.tsx
Comment (mmmmmmaya) [inline src/frontend/src/views/Upload/components/Metadata/index.tsx:60]: same type comment here

File: src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/utils.ts
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/utils.ts]: Why is this unknown?

File: src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/utils.ts
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/utils.ts]: can you break this out into a couple of lines so it's easier to read?

File: src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx]: please use a more descriptive var name here.

File: src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx]:
```suggestion
    // loses userinput if focus is taken away from the modal,
```

File: src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/utils.ts
Comment (phoenixAja) [reply src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/utils.ts]: changed

File: src/frontend/src/views/Upload/components/Metadata/index.tsx
Comment (phoenixAja) [reply src/frontend/src/views/Upload/components/Metadata/index.tsx:60]: this was mostly borrowed from the upload flow table, fixing the types isn't exactly straightforward so i added a TODO and hope to address it later.

Comment (phoenixAja) [reply src/frontend/src/views/Upload/components/Metadata/index.tsx:60]: i moved applyToAllColumn and handleRowMetadata up to the parent components of webformtable to better handle tracking the changes to metadata

File: src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx:117]: https://app.shortcut.com/genepi/story/193877/resolve-crud-v1-type-todo

Comment (phoenixAja) [reply src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx:117]: thanks for making the ticket!

**Comment by phoenixAja** [comment]
@mmmmmmaya anything else i need to resolve before getting approval?

**Review by mmmmmmaya — APPROVED** [review]
btw, it looks like you force pushed since I reviewed the PR. In principle there isn't anything wrong with that. The downside is, if you force push I cannot view unique changes since the last time I reviewed. That means that I effectively have to review the entire PR again because I can't isolate which changes were new vs old. 

On smaller ones it doesn't matter as much, but it has a larger impact on larger PR that are more logic heavy or have a lot of lines changed.

**Comment by phoenixAja** [comment]
@mmmmmmaya normally i wouldn't do that, but since i also have the tsv import branch i'm working off of thats branched from this i was trying to keep things as tidy as possible and avoid merge/rebase hell as much as possible since we're both working on a relatively similar spot in the code base, i'll try to do this minimally next time

**Pull request merged**
